Mathlib/Algebra/Polynomial/Roots.lean

```diff
@@ -612,11 +612,20 @@ lemma exists_eval_ne_zero_of_natDegree_lt_card (f : R[X]) (hf : f ≠ 0) (hfR :
 theorem monic_prod_multiset_X_sub_C : Monic (p.roots.map fun a => X - C a).prod :=
   monic_multiset_prod_of_monic _ _ fun a _ => monic_X_sub_C a
 
+theorem monic_finset_prod_X_sub_C [DecidableEq R] :
+    Monic (p.roots.toFinset.prod fun a => (X - C a)) :=
```

Comment:
This looks strange to me. Why do you need to consider this polynomial?

Reply:
I was asked to add this above by @alreadydone (see [this comment](https://github.com/leanprover-community/mathlib4/pull/23252#discussion_r2024735424)). Then I can use the multiset version of this lemma to conclude the  (now deleted) lemma `monic_of_eq_multiset_prod` from my first commit, which I use in the proof of the theorem [max_norm_root_eq_spectralValue]( https://github.com/leanprover-community/mathlib4/blob/93ae61b28bd4d73b02b6e916e8a4b04b91a9e9f5/Mathlib/Analysis/Normed/Unbundled/SpectralNorm.lean#L268) (if `f` is a nonarchimedean, power-multiplicative `K`-algebra norm on `L`, then the spectral value of a polynomial `p : K[X]` that decomposes into linear factos in `L` is equal to the maximum of the norms of the roots).

Reply:
This is the product of $X - r_i$ where $r_i$ runs through all the **unique** roots, i.e the squarefree-part of a polynomial. I just want to make sure if this is what you really want?

Reply:
I could delete this, I only really want the `Multiset` lemma (but maybe we should see what @alreadydone thinks, since he asked me to add this version).